File: 4824-2821-345-reverse-vowels-of-a-string/4824-2821-345-reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    const vowels = 'aeiouAEIOU';
    let i=0,j=s.length-1;
    let arr = s.split('');

    while(i<j){
        while(i<arr.length && !vowels.includes(arr[i])) i++;
        while(j>-1 && !vowels.includes(arr[j])) j--;

        if(i<j){
            [arr[i],arr[j]] = [arr[j],arr[i]];
            i++;
            j--;
        }
    }
    return arr.join('');
};

File: 4824-2821-345-reverse-vowels-of-a-string/4824-2821-345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        StringBuilder sb = new StringBuilder(s);
        String vowels = "AEIOUaeiou";
        int i=0,j=s.length()-1;
        while(i<j){
            while(i<s.length() && vowels.indexOf(sb.charAt(i))==-1)
                i++;
            while(j>0 && vowels.indexOf(sb.charAt(j))==-1)
                j--;
            if(i<j){
                char temp = sb.charAt(i);
                sb.setCharAt(i,sb.charAt(j));
                sb.setCharAt(j,temp);
                i++;
                j--;
            }
        }
        return sb.toString();
    }
}